bibliopixel/drivers/serial/driver.py

```diff
@@ -33,7 +32,7 @@ def __init__(self, ledtype=None, num=0, dev="",
 
         self._spi_speed = spi_speed
         self._com = None
-        self._ledtype = resolve_enum(LEDTYPE, ledtype)
```

Comment:
No no.... we're totally screwing over anyone who wants to write straight code. Now you cannot do `SPI(ledtype="APA102")`. You now only get this resolution if you use projects.

Reply:
I'm going to cobble something together to replace this, in the interests of backwards compatibility.

It's pretty lame, though.  I mean, we resolve exactly this one type of variable if you use "straight code" and no other?  Look at all those conversions in types - there are about twenty there, and _one_ of them works in "straight code" and none of the others.

If you are writing code, you could just write `LEDTYPE.APA102`.

At this point there is no rational reason to "write straight code" at all, and lots of reasons _not_ to - the conversions in types, all the loady stuff, but most important - if you write "straight" code you won't be able to share it with people, because your specific hardware is hardcoded into it!

Back to you in a bit with a revision.